sky/skylet/job_lib.py

```diff
@@ -341,21 +341,15 @@ def update_job_status(job_owner: str,
 
     job_client = _create_ray_job_submission_client()
 
-    def get_job_status(job_id) -> Optional[JobStatus]:
-        try:
-            # The return value is a string, e.g. 'RUNNING', which conflicts
-            # with the return type in ray code.
-            ray_status = job_client.get_job_status(job_id)
-            return _RAY_TO_JOB_STATUS_MAP[ray_status]
-        except RuntimeError as e:
-            # If the job does not exist or if the request to the
-            # job server fails.
-            if 'does not exist' in str(e):
-                return None
-            raise
+    ray_job_infos = job_client.list_jobs()
```

Comment:
Do you have a sense of the speedup due to this PR, perhaps add to PR description?

I was assuming the max concurrent jobs is single digit, but there can be lots of INIT jobs?

https://github.com/skypilot-org/skypilot/blob/5addab7b67428a6c2d02132d3c2672da4576c451/sky/skylet/job_lib.py#L420

Suggestion: update this variable to say `non_terminal_jobs`; was confused a bit by "running" (which would be single digit and thus doesn't need this change?).

Reply:
Good point! Renamed the variable.

For the speedup, I added it in the description. The original time is not that large (reduced by 3x), but the list_jobs can reduce the number of http requests by the number of jobs to be queried.